### PR TITLE
Add model type to client models.xml; various model apis also return model type

### PR DESCRIPTION
**Pull request by wallyworld**
## Description of change

A largely mechanical PR, we add model-type to the client models.xml file. This is so that various CLI commands can find out what type of model they are operating on.
Various model API calls need then to include model type in their result. This is included in show/list model output.
We also rename add-caas to add-k8s

## QA steps

bootstrap
check list/show model for controller and default models
create a caas model
check list/show model
verify content of models.xml file

### CODE REVIEW

File: api/controller/legacy_test.go [hunk 1]
```diff
@@ -9,6 +9,7 @@ import (
 	"net"
 	"time"
 
+	"github.com/juju/juju/core/model"
```
Comment (anastasiamac) [inline api/controller/legacy_test.go]: should be in the next block...

File: cmd/juju/commands/bootstrap.go [hunk 2]
```diff
@@ -13,6 +13,7 @@ import (
 	"github.com/juju/cmd"
 	"github.com/juju/errors"
 	"github.com/juju/gnuflag"
+	"github.com/juju/juju/core/model"
```
Comment (anastasiamac) [inline cmd/juju/commands/bootstrap.go]: move me

**Comment by wallyworld** [comment]
$$merge$$

**Comment by wallyworld** [comment]
$$merge$$

**Comment by wallyworld** [comment]
$$merge$$

File: jujuclient/validation.go [hunk 3]
```diff
@@ -21,6 +21,9 @@ func ValidateModelDetails(details ModelDetails) error {
 	if details.ModelUUID == "" {
 		return errors.NotValidf("missing uuid, model details")
 	}
+	if details.ModelType == "" {
```
Comment (rogpeppe) [inline jujuclient/validation.go]: This breaks the client when it's talking to an old API version which doesn't return a model type. (e.g. juju models fails like this:

    % juju models
    missing type, model details not valid
    ERROR missing type, model details not valid

Comment (wallyworld) [reply jujuclient/validation.go]: The api layer is supposed to fill in any missing model type values from older servers. If I have missed a place, that's a bug for sure.

**Pull request merged**